### ISSUE TITLE
Helm chart generates wrong --node-group-auto-discovery command line option on AWS

### ISSUE THREAD

**Issue by brunzefb** [kubernetes/autoscaler#4637]
Running on AWS.  My values file has the following:
```
autoDiscovery: 
  clusterName: [redacted]
  tags:
  - k8s.io/cluster-autoscaler/enabled
  - k8s.io/cluster-autoscaler/[redacted]
```

When deploying, the system generates the following:
```
# Source: cluster-autoscaler/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app.kubernetes.io/instance: "aws-cluster-autoscaler"
    app.kubernetes.io/name: "aws-cluster-autoscaler"
    app.kubernetes.io/managed-by: "Helm"
    helm.sh/chart: "cluster-autoscaler-9.9.2"
  name: aws-cluster-autoscaler
spec:
  replicas: 2
  selector:
    matchLabels:
      app.kubernetes.io/instance: "aws-cluster-autoscaler"
      app.kubernetes.io/name: "aws-cluster-autoscaler"
  template:
    metadata:
      annotations:
        iam.amazonaws.com/role: arn:aws:iam::[redacted]:role/AmazonEKSClusterAutoscalerRole-e3jdr
      labels:
        app.kubernetes.io/instance: "aws-cluster-autoscaler"
        app.kubernetes.io/name: "aws-cluster-autoscaler"
    spec:
      containers:
        - name: aws-cluster-autoscaler
          image: "k8s.gcr.io/autoscaling/cluster-autoscaler:v1.19.1"
          imagePullPolicy: "IfNotPresent"
          command:
            - ./cluster-autoscaler
            - --cloud-provider=aws
            - --namespace=kube-system
            - --node-group-auto-discovery=asg:tag=k8s.io/cluster-autoscaler/enabled, k8s.io/cluster-autoscaler/[redacted]   <== ISSUE WITH SPACE HERE
            - --balance-similar-node-groups=true
            - --expander=random
```

 Note that in the above, the --node-group-auto-discovery (third last line) has a space, which will cause the Autoscaler to not work (because it creates an invalid command line).

My solution was to edit the deployment.yaml, line 57 to remove the space using the replace function.
`- --node-group-auto-discovery=asg:tag={{ tpl (join "," .Values.autoDiscovery.tags) . | replace " " "" }} `

**Comment by brunzefb** [comment]
I could be persuaded to create a PR, if you ask nicely :-)

**Comment by sarahhodne** [comment]
I tried to replicate this, but I was unable to:

```YAML
% cat custom-values.yaml
autoDiscovery:
  clusterName: [redacted]
  tags:
  - k8s.io/cluster-autoscaler/enabled
  - k8s.io/cluster-autoscaler/[redacted]
```

```YAML
% helm template cluster-autoscaler . -f custom-values.yaml -s templates/deployment.yaml
---
# Source: cluster-autoscaler/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app.kubernetes.io/instance: "cluster-autoscaler"
    app.kubernetes.io/name: "aws-cluster-autoscaler"
    app.kubernetes.io/managed-by: "Helm"
    helm.sh/chart: "cluster-autoscaler-9.9.2"
  name: cluster-autoscaler-aws-cluster-autoscaler
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/instance: "cluster-autoscaler"
      app.kubernetes.io/name: "aws-cluster-autoscaler"
  template:
    metadata:
      labels:
        app.kubernetes.io/instance: "cluster-autoscaler"
        app.kubernetes.io/name: "aws-cluster-autoscaler"
    spec:
      dnsPolicy: "ClusterFirst"
      containers:
        - name: aws-cluster-autoscaler
          image: "k8s.gcr.io/autoscaling/cluster-autoscaler:v1.20.0"
          imagePullPolicy: "IfNotPresent"
          command:
            - ./cluster-autoscaler
            - --cloud-provider=aws
            - --namespace=default
            - --node-group-auto-discovery=asg:tag=k8s.io/cluster-autoscaler/enabled,k8s.io/cluster-autoscaler/[redacted]
            - --logtostderr=true
            - --stderrthreshold=info
            - --v=4
          env:
            - name: AWS_REGION
              value: "us-east-1"
          livenessProbe:
            httpGet:
              path: /health-check
              port: 8085
          ports:
            - containerPort: 8085
          resources:
            {}
      serviceAccountName: cluster-autoscaler-aws-cluster-autoscaler
      tolerations:
        []
```

The only way I was able to get a space in there was to insert some non-breaking spaces in the YAML, but even then I had to double up the spaces:

```YAML
autoDiscovery: 
  clusterName: [redacted]
  tags:
  - k8s.io/cluster-autoscaler/enabled
  -  k8s.io/cluster-autoscaler/[redacted]
#   ^ this is actually U+00A0
```

Assuming you don't have that in your values, which version of Helm are you using? That's the only thing I can think of that's different.

**Comment by brunzefb** [comment]
Hi,

I was using helm 3.8.0.  Verified that there are no weird control chars in the values file.   I then reset line 57 of deployment.yaml to 
- --node-group-auto-discovery=asg:tag={{ tpl (join "," .Values.autoDiscovery.tags) . }}
(with out the replace the space with nothing).

ran helm template, and there were no spaces after the comma in the command line args.

This is *very* weird -- I did not imagine this -- I remember editing the deployment with kubectl edit, and definitely there was a space after the comma, and the autoscaler did not autoscale.

Lets close the issue.  Thanks for taking the time to investigate.
F.
